### PR TITLE
Removed feature flag get-documents-from-ap-delius.

### PR DESCRIPTION
**Pull request by vbala-moj**
PR#1 for [https://dsdmoj.atlassian.net/browse/APS-1650](url)

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationDocumentsTest.kt [hunk 1]
```diff
@@ -249,9 +84,7 @@ class ApplicationDocumentsTest : InitialiseDatabasePerClassTestBase() {
   }
 
   @Test
-  fun `Download document - with get-documents-from-ap-delius feature-flag on - returns 200 with correct body and headers`() {
-    mockFeatureFlagService.setFlag("get-documents-from-ap-delius", true)
-
+  fun `Download document - returns 200 with correct body and headers`() {
```
Comment (davidatkinsuk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationDocumentsTest.kt:87]: perfect, good that we've renamed these 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt [hunk 2]
```diff
@@ -407,31 +404,20 @@ class ApplicationsController(
...
-  private fun getDocuments(isDocumentFromAPDelius: Boolean, crn: String) = if (isDocumentFromAPDelius) {
-    offenderService.getDocumentsFromApDeliusApi(crn)
-  } else {
-    offenderService.getDocumentsFromCommunityApi(crn)
+  private fun getDocuments(crn: String): List<APDeliusDocument> {
+    return when (val result = offenderService.getDocumentsFromApDeliusApi(crn)) {
+      is AuthorisableActionResult.NotFound -> throw NotFoundProblem(crn, "Documents")
+      is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
+      is AuthorisableActionResult.Success -> result.entity
+    }
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt:420]: nice, i endorse moving this logic into the private method 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt [hunk 3]
```diff
@@ -407,31 +404,20 @@ class ApplicationsController(
...
 
-    val documentsFromApDeliusFlag = featureFlagService.getBooleanFlag("get-documents-from-ap-delius")
-    val documents = when (val documentsResult = getDocuments(isDocumentFromAPDelius = documentsFromApDeliusFlag, application.crn)) {
-      is AuthorisableActionResult.NotFound -> throw NotFoundProblem(application.crn, "Person")
-      is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
-      is AuthorisableActionResult.Success -> documentsResult.entity
-    }
-
-    val transformedDocuments = when (documents) {
-      is GroupedDocuments -> documentTransformer.transformToApi(documents)
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt:421]: I think this function on DocumentTransformer can now be removed as the only thing that should be using it is unit tests:

`fun transformToApi(groupedDocuments: GroupedDocuments)`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt [hunk 4]
```diff
@@ -407,31 +404,20 @@ class ApplicationsController(
...
+    val apiDocuments = documentTransformer.transformToApi(documentsResult)
 
-    return ResponseEntity(transformedDocuments, HttpStatus.OK)
+    return ResponseEntity(apiDocuments, HttpStatus.OK)
   }
 
-  private fun getDocuments(isDocumentFromAPDelius: Boolean, crn: String) = if (isDocumentFromAPDelius) {
-    offenderService.getDocumentsFromApDeliusApi(crn)
-  } else {
-    offenderService.getDocumentsFromCommunityApi(crn)
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt:434]: I think this function can now be removed?

`fun getDocumentsFromCommunityApi(crn: String): AuthorisableActionResult<GroupedDocuments>`

**Review by davidatkinsuk — APPROVED** [review]
Looks great, just a few proposals from me on some code that can be removed 👍

**Comment by vbala-moj** [comment]
> Looks great, just a few proposals from me on some code that can be removed 👍

Sure, will remove those methods on the next PR.

**Pull request merged**